doc/intro-to-autoinstall.rst

```diff
@@ -38,23 +38,22 @@ Providing the autoinstall configuration
 =======================================
 
 There are 2 ways to provide the autoinstall configuration:
- * Provide :external+cloud-init:ref:`#cloud-config
-   user-data<user_data_formats-cloud_config>` containing ``autoinstall:``
+ * Provide :external+cloud-init:ref:`#cloud-configuser-data<user_data_formats-cloud_config>` containing ``autoinstall:``
```

Comment:
This doesn't look quite correct.  Let's compare to the cloud-init docs and pick something similar.  Call it `Cloud config data` in the displayed text?

![image](https://github.com/canonical/subiquity/assets/235157/92d109ed-5fb1-4eb4-877a-a1420d5324ea)

Reply:
I shoud've checked the linked docs... The key is called `#cloud-config`, so let's use that. "Cloud config" doesn't look good.